src/hardware/driver/dac.rs

```diff
@@ -180,24 +187,28 @@ where
 
         // reset DAC
         dac.write(DAC_ADDR::TRIGGER, TRIGGER::SRST::RESET);
+        delay.delay_us(200);
         // set to 10 V plusminus 1.25 V referenece span, retain defaults
         dac.write(
             DAC_ADDR::CONFIG1,
             CONFIG1::VREFVAL::SPAN_10V
-                | CONFIG1::DSDO::ENABLED // set to retain default
-                | CONFIG1::FSET::ENABLED // set to retain default
-                | CONFIG1::LDACMODE::ASYNC, // set to use SPI SYNC
+            | CONFIG1::DSDO::ENABLED // set to retain default
+            | CONFIG1::FSET::ENABLED // set to retain default
+            | CONFIG1::LDACMODE::ASYNC, // set to use SPI SYNC
         );
         // perform calibration
         // don't try to calibrate without driver because it will wait forever
         // dac.calibrate(delay);
-
+        delay.delay_us(200);
```

Comment:
I have found a delay of more than 10 us necessary here, otherwise the current setting in the next line does not work. The delay after the reset does not have an effect (even though its probaply better to give it some time after a reset) so it must be specifically after writing the CONFIG1 reg. I couldn't find this in the datasheet.. Everywhere else >100 ns delays are enough between writes as per datasheet.

Reply:
Probably a good idea to check the ALARM bit. It may well indicate that the DAC isn't ready yet.

Reply:
Yes, that is what is done in 'calibration()' once we have fixed the DAC readback in HW.

Reply:
Sure. I'd just keep track of it.